plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -1,19 +1,258 @@
 package com.google.cloud.tools.eclipse.appengine.deploy.ui;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Control;
 import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Text;
 import org.eclipse.ui.IWorkbenchPropertyPage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.browser.IWorkbenchBrowserSupport;
 import org.eclipse.ui.dialogs.PropertyPage;
+import org.eclipse.ui.forms.widgets.ExpandableComposite;
+
+import com.google.cloud.tools.eclipse.ui.util.FontUtil;
 
 public class DeployPropertyPage extends PropertyPage implements IWorkbenchPropertyPage {
 
+  private static final String APPENGINE_DASHBOARD_URL = "https://console.developers.google.com/appengine";
+  private static final int INDENT_CHECKBOX_ENABLED_WIDGET = 10;
+
+  private Button promptForProjectIdButton;
+  private Label projectIdLabel;
```

Comment:
If they're fields, it's easier to call `setEnabled()` on them. On Mac
there's not much difference, but on Linux they also greyed out, something
that's consistent with the other preference pages (i.e. disable labels as
well as input widgets).

On Thu, Aug 11, 2016, 07:02 Elliotte Rusty Harold <notifications@github.com>
wrote:

> In
> plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/529#discussion_r74402903>
> :
>
> >
> >  public class DeployPropertyPage extends PropertyPage implements IWorkbenchPropertyPage {
> >
> > +  private static final String APPENGINE_DASHBOARD_URL = "https://console.developers.google.com/appengine";
> > +  private static final int INDENT_CHECKBOX_ENABLED_WIDGET = 10;
> > +
> > +  private Button promptForProjectIdButton;
> > +  private Label projectIdLabel;
>
> I think the three labels can all be moved to local variables rather than
> fields
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/529/files/ab60fe8ba02bfe8a123a0d40213044ef4450d00b#r74402903>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADOO_qB9mBQtXyG-0in7wGotuEK2uD48ks5qewEdgaJpZM4Jgjkk>
> .
>